webapp/src/index.tsx

```diff
@@ -413,10 +422,23 @@ export default class Plugin {
         };
 
         const fetchChannelData = async (channelID: string) => {
-            const channel = getChannel(store.getState(), channelID);
+            let channel = getChannel(store.getState(), channelID);
+            if (!channel) {
+                await getChannelAction(channelID)(store.dispatch as any, store.getState);
+                channel = getChannel(store.getState(), channelID);
+            }
+
             const roles = getMyRoles(store.getState());
             const cms = getMyChannelMemberships(store.getState());
 
+            if (isDMChannel(channel)) {
+                const otherID = getUserIdFromDM(channel.name, getCurrentUserId(store.getState()));
+                const dmUser = getUser(store.getState(), otherID);
+                if (!dmUser) {
+                    await getProfilesByIdsAction([otherID])(store.dispatch as any, store.getState);
```

Comment:
Not a fan of using `any` but I hit this issue before and my still fairly limited Typescript/React knowledge couldn't get to an easy solution.

Error was the following:

```
src/index.tsx:438:61 - error TS2345: Argument of type 'Dispatch<AnyAction>' is not assignable to parameter of type 'DispatchFunc'.
  Types of parameters 'action' and 'action' are incompatible.
    Type 'Action' is not assignable to type 'AnyAction'.
      Property 'type' is missing in type 'ActionFunc' but required in type 'AnyAction'.

438                     await getProfilesByIdsAction([otherID])(store.dispatch, store.getState);
                                                                ~~~~~~~~~~~~~~

  node_modules/redux/index.d.ts:19:3
    19   type: T
         ~~~~
    'type' is declared here.
```

Reply:
Do you mind if I push a commit? It will add `redux-thunk` to the dev dependencies and simplify this a bit?

Reply:
By all means :raised_hands: